### PR TITLE
Build RPM in pre-merge CI

### PR DESCRIPTION
**Pull request by ncoghlan**
Also fixes current tito build errors

### CODE REVIEW

File: centos-ci/bootstrap.sh [hunk 1]
```diff
@@ -12,6 +12,11 @@ yum install -y python2-pip gcc redhat-rpm-config openssl-devel python-devel
 # Interim EPEL-based approach to enable testing of --ask-pass option
 yum install -y sshpass
 
+# Check the RPM can be built successfully
+yum install tito
```
Comment (vinzenz) [inline centos-ci/bootstrap.sh]: missing -y this will not work - AFAICT

**Comment by ncoghlan** [comment]
The CI stalled because of another missing `-y` option, so I took the opportunity to clean up and rebase the PR while fixing that.

**Comment by ncoghlan** [comment]
It turns out the cryptic "no job control message" almost certainly means RPM on CentOS 7 isn't recognising the `%py2_*` macros as valid, and hence the `%` is getting passed through to the shell.

**Comment by ncoghlan** [comment]
The RPM build is working now, but there was a bug in the operand precedence for the build command that caused the bootstrap script to always exit immediately, even when the build succeeded.

I've fixed that, and also tidied up the commit history again.

**Pull request merged**